formats/label-to-vector-plugin/src/polus/plugins/formats/label_to_vector/dynamics/label_to_vector.py

```diff
@@ -0,0 +1,419 @@
+    axes = tuple(range(1, stack.ndim))
+
+    vector_slices = list(numpy.gradient(heat, axis=axes))
+    vector_slices = numpy.stack(vector_slices, axis=-1)
+    vector_slices = common.vector_norm(vector_slices)
+    vector_slices[~stack] = 0
+
+    return vector_slices
+
+
+def gradient_store(  # noqa: PLR0913
+    stack: numpy.ndarray,
+    slices: list[tuple[slice, slice]],
+    sub_slices: list[tuple[slice, slice]],
+    centroid: numpy.ndarray,
+    sc: int,
+    output: numpy.ndarray,
+) -> int:
+    """Calculate vector fields.
+
+    This function is primarily designed to be run inside a thread, otherwise this code
+    would just belong in the `labels_to_vectors` function.
+
+    Args:
+        stack: Stack of image masks
+        slices: A list of slice tuples indicating the location of a particular
+            mask in the original image.
+        sub_slices: A list of slice tuples that indicate the bounds of a mask
+            within the stack.
+        centroid: The location of the centroid in the stack.
+        sc: The starting roi ID.
+        output: The output image vector field
+
+    Returns:
+        The number of ROIs processed
+    """
+    heat_slices = creeping_mean_filter(stack, centroid, sc)
+
+    vector_slices = _find_gradients(stack, heat_slices)
+
+    for i, s in enumerate(vector_slices):
+        output[slices[sc + i]] += s[sub_slices[i]]
+
+    return stack.shape[0]
+
+
+def convert(
+    masks: numpy.ndarray,
+    name: typing.Optional[str] = None,
+) -> numpy.ndarray:
+    """Convert labels to vector fields.
+
+    This function uses a modified diffusion algorithm to calculate vector flow fields
+    from the centroid of an object out to its edges.
+
+    Args:
+        masks: An image in 2d or 3d
+        name: Name of the image to process (for logging purposes). Defaults to None.
+
+    Returns:
+        An ND Array of vectors, where N is masks.ndims+1, where the first dimension
+            are different components of the vectors. For example, for a 2D
+            input mask of size MxN, the output shape will be (2,M,N).
+    """
+    name = (name + ": ") if name is not None else ""
+
+    if masks.ndim not in [2, 3]:
+        message = f"{name}Masks must be 2d or 3d. Got {int(masks.ndim)}d instead."
+        logger.error(message)
+        raise ValueError(message)
+
+    # ensure unique and contiguous labels
+    uniques, inverse = numpy.unique(masks, return_inverse=True)
+    if len(uniques) == 1:
+        if uniques[0] == 0:
+            logger.warning(f"{name} No objects found. Returning zero flows.")
+        else:
+            logger.warning(
+                f"{name} The masked tile/chunk contains one object and no "
+                "background pixels. Returning zero flows.",
+            )
+        return numpy.zeros(shape=(masks.ndim, *masks.shape), dtype=numpy.float32)
+
+    labels = numpy.reshape(inverse, masks.shape)
+    slices = list(scipy.ndimage.find_objects(labels))
+
+    sc = 0
```

Comment:
slice count? IMO this is not a good variable name since intent is not obvious at all